network_simulator/link.py

```diff
@@ -95,3 +159,117 @@ def __eq__(self, other):
 
     def __ne__(self, other):
         return not self.__eq__(other)
+
+    def get_link_cost(self):
+        """
+        Returns link cost. For now, just returns static propagation delay.
+        :return: float of link delay time in seconds
+        """
+
+        # TODO consider average wait time for last N packets
+        return self.static_delay_sec
+
+    def push(self, packet, dest_device_addr, global_clock_sec):
+        """
+        :param Packet packet: Packet to push
+        :param string dest_device_addr: destination device of packet
+        :param global_clock_sec: store when packet pushed
+
+        Pushes packet onto own LinkBuffer going towards dest_device_addr
+        :return: True if pushed, False if not
+        """
+        return self.link_buffer.push(packet, dest_device_addr, global_clock_sec)
+
+class LinkSendEvent(Event):
+    """
+    Event representing a packet arriving at the Link
+    """
+
+    def __init__(self, link):
+        """
+        :ivar link: link that must send its next packet
+        """
+        self.link = link
+        self.buffer_elem = None
+    def run(self, main_event_loop, statistics):
+        """
+        runs by popping off top packet
+        :param MainEventLoop main_event_loop: event loop for global timing
+        :param Statistics statistics: the Statistics to update
+        """
+
+        # sanity check
+        if(self.link.queue.empty()):
+            logger.warning("LinkSendEvent while Link empty!")
```

Comment:
Do we have a logger class imported? Does this compile?
LOL sorry I don't know if we do.